server/utils/firebase_algolia_server_code/functions/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Description: Turn ad entities into algolia records
+ *              With HTTP cloud functions  
+ *              Basically compile with `firebase deploy --only functions`
+ *              These will then appear in functions tab of firestore 
+ * Author: Robert Marcus
+ * Date: July 7, 2020
+ */
+
+const { algoliaFunctions, INDEX, DOCS } = require('./indexConfig');
+
+/**
+ * Cloud functions below
+ * Vocab: a firestore collection = algolia index
+ *        an ad entity = algolia record
+ * Note: these are background events, basically event listeners
+ * They are guaranteed 'at-least-once execution'
+ * If these functions fail, they will not attempt to retry (currently)
+ * Firebase has native support for enabling retry on failure
+ * This feature will be integrated at some point post-MVP. 
+ * https://cloud.google.com/functions/docs/bestpractices/retries
+ */
+
+/**
+ * Creates an algolia record from a firebase entity snapshot.
+ * Use default value unless mocking an algoliaOperation.
+ * @param {function(string): !Promise=} algoliaOperation a save function
+ * @return {function} 
+ */
+function createRecordFromEntity(algoliaOperation = INDEX.saveObject) {
+  exports.create = 
```

Comment:
Firebase convention for deploying cloud function `bar` is to write it as `exports.bar`. Unfortunately this is a little confusing as you've noted. The `module.exports` is used to export the function wrappers for testing, and is unrelated to the `exports` within the closures. I've actually posted to stack exchange about this because I find it confusing and I want to see if their is a better way to do this.